### PR TITLE
fix: support POST-only API of go-ipfs 0.5

### PR DESCRIPTION
**Pull request by lidel**
## TL;DR

This PR ensures orbit-db remains compatible with HTTP API exposed by go-ipfs 0.5 (https://github.com/ipfs/go-ipfs/issues/7109) by switching to `ipfs-http-client@41.0.1` in tests and adding interop note in README.  

## Changes

- **added interop note to README about compatible http client version**
  - note about version limitation can be removed when #767 lands
- bumped versions of http-client and go-ipfs used in tests to ensure it works with latest go-ipfs 0.5.0-rc2

## Why?

go-ipfs 0.5  will block `GET` commands on the API port (https://github.com/ipfs/go-ipfs/pull/7097), requiring every  command (RPC) to be sent as HTTP `POST` request.

Why v41.0.1? It sends only POST requests while maintaining the old JS API 
(breaking changes that introduced support for async iterators landed in 42.x)

cc @aphelionz @stebalien @achingbrain

### CODE REVIEW

**Comment by aphelionz** [comment]
Thank you @lidel!!

**Comment by Stebalien** [comment]
The go-ipfs 0.5.0 release will be cut later today. Any chance this can be merged soon?

**Comment by aphelionz** [comment]
@Stebalien Merged to master for now, before our own release. Hope that's helpful!

**Comment by Stebalien** [comment]
It does. Thanks!

**Comment by aphelionz** [comment]
@lidel @Stebalien should the practice org-wide in `orbitdb` be for us to utilize `ipfs-http-client@~41.0.1` and `go-ipfs@0.5.0-rc2`?

**Comment by lidel** [comment]
Yes (0.5.0 just shipped) :)

**Comment by Stebalien** [comment]
Is there any chance we could cut a patch release with this change so users can upgrade go-ipfs without migrating to async/await?
